components/nimbus/tests/common/mod.rs

```diff
@@ -7,13 +7,25 @@ use nimbus::{error::Result, AppContext, NimbusClient, RemoteSettingsConfig};
 
 #[allow(dead_code)] // not clear why this is necessary...
 pub fn new_test_client(identifier: &str) -> Result<NimbusClient> {
-    use std::path::PathBuf;
     use tempdir::TempDir;
+    let tmp_dir = TempDir::new(identifier)?;
+
+    new_test_client_internal(&tmp_dir)
+}
+
+#[allow(dead_code)] // not clear why this is necessary...
```

Comment:
`[allow(dead_code)]` is needed here as a workaround for https://github.com/rust-lang/rust/issues/46379

Perhaps you could update the `// not clear why this is necessary` comments and redirect to the rust bug/issue?

Reply:
Done for all experiments in this file.